src/main/java/org/apache/sysml/hops/ipa/FunctionCallGraph.java

```diff
@@ -64,8 +68,9 @@
 	
 	//subset of direct or indirect recursive functions
 	private final HashSet<String> _fRecursive;
-	
-	private final boolean _containsEval;
+
+	// a boolean value to indicate if exists the second order function (e.g. eval, paramserv)
+	private final boolean _containsSecondOrder;
```

Comment:
yes, that is a better name - could you please extend the comment to indicate that this also covers UDFs that are marked `secondorder="true"`.